### ISSUE TITLE
Test failure on install

### ISSUE THREAD

**Issue by jnthn** [libxml-raku/LibXML-raku#36]
I'm seeing the following test failure when trying to install this module:

```
===> Testing: LibXML:ver<0.3.1>:auth<cpan:WARRINGD>
[LibXML] # Running libxml2 version: 2.09.03 (module 0.3.1)
[LibXML] # Failed test 'close1 for $parser out of ($parser,$parser2)'
[LibXML] # at /home/jnthn/.zef/store/LibXML-0.3.1.tar.gz/LibXML-0.3.1/t/lib/Counter.pm6 (Counter) line 13
[LibXML] # expected: '1'
[LibXML] #      got: '2'
[LibXML] # You failed 1 test of 25
===> Testing [FAIL]: LibXML:ver<0.3.1>:auth<cpan:WARRINGD>
Aborting due to test failure: LibXML:ver<0.3.1>:auth<cpan:WARRINGD> (use --force-test to override)
```

I'm on Ubuntu 16.04 and just installed `libxml2-dev`, so figure it's a decently recent version. I'm running a recent Rakudo build ("This is Rakudo version 2020.05.1-63-g9a583dc built on MoarVM version 2020.05-2-g60b5670 implementing Raku 6.d."). If I can provide any further information to help fix it, please do let me know.

**Comment by dwarring** [comment]
LibXML actually invokes the close() callback when it's cleaning up in xmlFreeParserCtxt() which is don't get called until LibXML::Parser::Context.DESTROY() is invoked.

So the counting of close() calls in t/17-callbacks.t is unreliable.

**Comment by dwarring** [comment]
These fixes add and use method LibXML::XPath::Context close() to destroy the input buffer as soon as parsing has finished. Triggering the close callbacks reliably and earlier.

**Comment by dwarring** [comment]
Fixed in LibXML 0.4.0, uploaded to CPAN
